src/bin/commands/account-readiness/vpc.ts

```diff
@@ -0,0 +1,58 @@
+import AWS from "aws-sdk";
+import chalk from "chalk";
+import type { AwsAccountReadiness } from "../../../types/cli";
+import { doesDefaultVpcSsmParameterExist, getSsmParametersForVpc, getVpcsInDetail } from "../../../utils/cli/vpc";
+
+const UNKNOWN = "unknown";
+
+export const vpcReadiness = async ({ credentialProvider, region }: AwsAccountReadiness): Promise<number> => {
+  const ec2Client = new AWS.EC2({
+    credentialProvider,
+    region,
+  });
+  const ssmClient = new AWS.SSM({
+    credentialProvider,
+    region,
+  });
+
+  const vpcs = await getVpcsInDetail(ec2Client);
+  const inUseVpcs = vpcs.filter((vpc) => vpc.isUsed);
+  const ssmParamsFromAccount = await getSsmParametersForVpc(ssmClient);
+  const usingDeprecatedSsmParameter = doesDefaultVpcSsmParameterExist(ssmParamsFromAccount);
+
+  console.group(chalk.bold("\nVPC Summary"));
+
+  if (usingDeprecatedSsmParameter) {
+    console.log(
+      `❌ SSM parameters with path '/account/vpc/default/*' found. Please rename to '/account/vpc/primary/*'`
+    );
+  }
+
+  const totalSsmParametersExpected = inUseVpcs.length * 3;
```

Comment:
I'm wondering if this assumption is correct. Currently the CLI complains about the Deploy Tools account because we don't have _enough_ SSM parameters. I believe this is because the default VPC is technically in use at the moment, but it is not referenced in SSM? 

IIUC, we don't want teams to use the default VPC at all? However, the CLI output implies that to 'fix' things, I could quickly add SSM parameters which point to the default VPC. 

Although there is another potential action here (migrate everything away from the default VPC!), this is non-trivial and is also not strictly required in order to start using `@guardian/cdk` (in fact I might be able to use the library to help me to spin up new infrastructure in the correct VPC as part of a migration).

So perhaps it would be more appropriate to check:

* The number of SSM parameters == `inUseNonDefaultVpcs.length * 3`

And (for bonus points!):

* There are no `/account/vpc/*` SSM parameters that point to the default VPC

Reply:
> So perhaps it would be more appropriate to check:
>
> - The number of SSM parameters == inUseNonDefaultVpcs.length * 3

Yes I think this makes sense.

You've also reminded me of a further check that needs implementing:
  - If there are multiple VPCs in use, a param of `/account/vpc/primary/*` should not exist

I'm also not entirely happy by hard-coding `3` here and would prefer for it to be a `.length` of something. Will have a think.

Reply:
> You've also reminded me of a further check that needs implementing:
> 
> * If there are multiple VPCs in use, a param of `/account/vpc/primary/*` should not exist

I agree with this in principle, but in practice it's also possible that there are 2 VPCs currently in use but the team are trying to consolidate into one, so I'd be inclined to chat with teams about this on a case-by-case basis rather than encoding it as a strict rule.

Reply:
👍🏽  (not only because it's less work)

Reply:
> So perhaps it would be more appropriate to check:
> 
> - The number of SSM parameters == inUseNonDefaultVpcs.length * 3
>
> And (for bonus points!):
> 
> - There are no /account/vpc/* SSM parameters that point to the default VPC

Thinking about this a little more, this would mean accounts would need to migrate away from the default VPC first in order to adopt GuCDK. That's quite a lot of work! According to Prism, we have 11 accounts using the default VPC - 1/3 of all our accounts!

What if we log/warn if:
  - Default VPCs are used
  - SSM Parameters point to default VPCs

Reply:
Yes I think that's a sensible compromise 👍

Reply:
Updated now.